$NetBSD$

Update the two descriptor heap references for glslang 16.6 and upstream
47587f99b888. The current input represents readonly and writeonly images
in separate decorated blocks; the output must retain both access qualifiers.
The legacy mapping case also retains set 1, binding 2 and nonuniform indices.
Other generated names follow input IDs; both outputs compile and validate.

--- reference/shaders-no-opt/vulkan/frag/descriptor-heap.vk.nocompat.spv16.frag.vk.orig
+++ reference/shaders-no-opt/vulkan/frag/descriptor-heap.vk.nocompat.spv16.frag.vk
@@ -8,69 +8,72 @@
 
 layout(location = 0) out vec4 FragColor;
 rayQueryEXT rq;
-layout(descriptor_heap) uniform texture1D spv_23ResourceHeap[];
-layout(descriptor_heap) uniform texture2D spv_36ResourceHeap[];
-layout(descriptor_heap) uniform texture3D spv_49ResourceHeap[];
-layout(descriptor_heap) uniform sampler spv_71SamplerHeap[];
-layout(descriptor_heap, r32f) uniform image1D spv_108ResourceHeap[];
-layout(descriptor_heap, r32f) uniform image2D spv_115ResourceHeap[];
-layout(descriptor_heap, r32f) uniform image3D spv_122ResourceHeap[];
-layout(descriptor_heap) uniform image2D spv_150ResourceHeap[];
+layout(descriptor_heap) uniform texture1D spv_14ResourceHeap[];
+layout(descriptor_heap) uniform texture2D spv_32ResourceHeap[];
+layout(descriptor_heap) uniform texture3D spv_45ResourceHeap[];
+layout(descriptor_heap) uniform sampler spv_64SamplerHeap[];
+layout(descriptor_heap, r32f) writeonly uniform image1D spvWriteImages1D_heapResourceHeap[];
+layout(descriptor_heap, r32f) writeonly uniform image2D spvWriteImages2D_heapResourceHeap[];
+layout(descriptor_heap, r32f) writeonly uniform image3D spvWriteImages3D_heapResourceHeap[];
+layout(descriptor_heap, r32f) readonly uniform image1D spvRWriteImages1D_heapResourceHeap[];
+layout(descriptor_heap, r32f) readonly uniform image2D spvRWriteImages2D_heapResourceHeap[];
+layout(descriptor_heap, r32f) readonly uniform image3D spvRWriteImages3D_heapResourceHeap[];
+layout(descriptor_heap) uniform image2D spv_145ResourceHeap[];
 layout(descriptor_heap, std140) uniform UBO140
 {
     float data[2];
-} spvUBO140_167ResourceHeap[];
+} spvUBO140_164ResourceHeap[];
 
 layout(descriptor_heap, std140) uniform UBO430
 {
     vec3 data[2];
-} spvUBO430_179ResourceHeap[];
+} spvUBO430_177ResourceHeap[];
 
 layout(descriptor_heap, scalar) uniform UBOScalar
 {
     vec3 data[2];
-} spvUBOScalar_191ResourceHeap[];
+} spvUBOScalar_189ResourceHeap[];
 
-layout(descriptor_heap, std430) readonly buffer SSBOReadOnly
+layout(descriptor_heap, std430) readonly buffer SSBOReadOnlyNoWrite
 {
     vec4 data;
-} spvSSBOReadOnly_204ResourceHeap[];
+} spvSSBOReadOnlyNoWrite_202ResourceHeap[];
 
-layout(descriptor_heap, std430) coherent writeonly buffer SSBOWriteOnly
+layout(descriptor_heap, std430) coherent writeonly buffer SSBOWriteOnlyNoRead
 {
     vec4 data;
-} spvSSBOWriteOnly_216ResourceHeap[];
+} spvSSBOWriteOnlyNoRead_215ResourceHeap[];
 
 layout(descriptor_heap, std430) buffer SSBO
 {
     vec4 data;
-} spvSSBO_223ResourceHeap[];
+} spvSSBO_222ResourceHeap[];
 
-layout(descriptor_heap, r32ui) uniform uimage2D spv_229ResourceHeap[];
-layout(descriptor_heap) uniform accelerationStructureEXT spv_241ResourceHeap[];
+layout(descriptor_heap, r32ui) uniform uimage2D spv_227ResourceHeap[];
+layout(descriptor_heap) uniform accelerationStructureEXT spv_240ResourceHeap[];
 
 void main()
 {
     FragColor = vec4(0.0);
-    FragColor += texelFetch(spv_23ResourceHeap[int(gl_FragCoord.x)], 0, 0);
-    FragColor += texelFetch(spv_36ResourceHeap[int(gl_FragCoord.x)], ivec2(0), 0);
-    FragColor += texelFetch(spv_49ResourceHeap[int(gl_FragCoord.x)], ivec3(0), 0);
-    FragColor += texture(sampler2D(spv_36ResourceHeap[int(gl_FragCoord.x)], spv_71SamplerHeap[int(gl_FragCoord.y)]), vec2(0.0), 0.0);
-    FragColor += vec4(texture(sampler2DShadow(spv_36ResourceHeap[int(gl_FragCoord.x)], spv_71SamplerHeap[int(gl_FragCoord.y)]), vec3(vec3(0.0).xy, 0.0), 0.0));
-    imageStore(spv_108ResourceHeap[10], 0, FragColor);
-    imageStore(spv_115ResourceHeap[20], ivec2(0), FragColor);
-    imageStore(spv_122ResourceHeap[30], ivec3(0), FragColor);
-    FragColor += imageLoad(spv_108ResourceHeap[10], 0);
-    FragColor += imageLoad(spv_115ResourceHeap[20], ivec2(0));
-    FragColor += imageLoad(spv_122ResourceHeap[30], ivec3(0));
-    FragColor += imageLoad(spv_150ResourceHeap[40], ivec2(0));
-    FragColor += vec4(spvUBO140_167ResourceHeap[50].data[1]);
-    FragColor += vec4(spvUBO430_179ResourceHeap[51].data[1].x);
-    FragColor += vec4(spvUBOScalar_191ResourceHeap[52].data[1].x);
-    FragColor += spvSSBOReadOnly_204ResourceHeap[60].data;
-    spvSSBOWriteOnly_216ResourceHeap[61].data = vec4(20.0);
-    FragColor += spvSSBO_223ResourceHeap[62].data;
-    uint _237 = imageAtomicAdd(spv_229ResourceHeap[70], ivec2(0), 50u);
-    rayQueryInitializeEXT(rq, spv_241ResourceHeap[50], 0u, 0u, vec3(0.0), 0.0, vec3(1.0, 0.0, 0.0), 1.0);
+    FragColor += texelFetch(spv_14ResourceHeap[int(gl_FragCoord.x)], 0, 0);
+    FragColor += texelFetch(spv_32ResourceHeap[int(gl_FragCoord.x)], ivec2(0), 0);
+    FragColor += texelFetch(spv_45ResourceHeap[int(gl_FragCoord.x)], ivec3(0), 0);
+    FragColor += texture(sampler2D(spv_32ResourceHeap[int(gl_FragCoord.x)], spv_64SamplerHeap[int(gl_FragCoord.y)]), vec2(0.0), 0.0);
+    FragColor += vec4(texture(sampler2DShadow(spv_32ResourceHeap[int(gl_FragCoord.x)], spv_64SamplerHeap[int(gl_FragCoord.y)]), vec3(vec3(0.0).xy, 0.0), 0.0));
+    imageStore(spvWriteImages1D_heapResourceHeap[10], 0, FragColor);
+    imageStore(spvWriteImages2D_heapResourceHeap[20], ivec2(0), FragColor);
+    imageStore(spvWriteImages3D_heapResourceHeap[30], ivec3(0), FragColor);
+    FragColor += imageLoad(spvRWriteImages1D_heapResourceHeap[10], 0);
+    FragColor += imageLoad(spvRWriteImages2D_heapResourceHeap[20], ivec2(0));
+    FragColor += imageLoad(spvRWriteImages3D_heapResourceHeap[30], ivec3(0));
+    FragColor += imageLoad(spv_145ResourceHeap[40], ivec2(0));
+    FragColor += vec4(spvUBO140_164ResourceHeap[50].data[1]);
+    FragColor += vec4(spvUBO430_177ResourceHeap[51].data[1].x);
+    FragColor += vec4(spvUBOScalar_189ResourceHeap[52].data[1].x);
+    FragColor += spvSSBOReadOnlyNoWrite_202ResourceHeap[60].data;
+    spvSSBOWriteOnlyNoRead_215ResourceHeap[61].data = vec4(20.0);
+    FragColor += spvSSBO_222ResourceHeap[62].data;
+    uint _236 = imageAtomicAdd(spv_227ResourceHeap[70], ivec2(0), 50u);
+    rayQueryInitializeEXT(rq, spv_240ResourceHeap[50], 0u, 0u, vec3(0.0), 0.0, vec3(1.0, 0.0, 0.0), 1.0);
 }
 
--- reference/shaders-no-opt/vulkan/frag/descriptor-heap.heap-legacy-mapping.vk.nocompat.spv16.frag.vk.orig
+++ reference/shaders-no-opt/vulkan/frag/descriptor-heap.heap-legacy-mapping.vk.nocompat.spv16.frag.vk
@@ -8,76 +8,79 @@
 
 layout(location = 0) out vec4 FragColor;
 rayQueryEXT rq;
-layout(set = 1, binding = 2) uniform texture1D spv_28ResourceHeap[];
-layout(set = 1, binding = 2) uniform texture2D spv_40ResourceHeap[];
-layout(set = 1, binding = 2) uniform texture3D spv_53ResourceHeap[];
-layout(set = 1, binding = 2) uniform sampler spv_75SamplerHeap[];
-layout(set = 1, binding = 2, r32f) uniform image1D spv_114ResourceHeap[];
-layout(set = 1, binding = 2, r32f) uniform image2D spv_123ResourceHeap[];
-layout(set = 1, binding = 2, r32f) uniform image3D spv_132ResourceHeap[];
-layout(set = 1, binding = 2) uniform image2D spv_171ResourceHeap[];
+layout(set = 1, binding = 2) uniform texture1D spv_25ResourceHeap[];
+layout(set = 1, binding = 2) uniform texture2D spv_36ResourceHeap[];
+layout(set = 1, binding = 2) uniform texture3D spv_49ResourceHeap[];
+layout(set = 1, binding = 2) uniform sampler spv_68SamplerHeap[];
+layout(set = 1, binding = 2, r32f) writeonly uniform image1D spvWriteImages1D_heapResourceHeap[];
+layout(set = 1, binding = 2, r32f) writeonly uniform image2D spvWriteImages2D_heapResourceHeap[];
+layout(set = 1, binding = 2, r32f) writeonly uniform image3D spvWriteImages3D_heapResourceHeap[];
+layout(set = 1, binding = 2, r32f) readonly uniform image1D spvRWriteImages1D_heapResourceHeap[];
+layout(set = 1, binding = 2, r32f) readonly uniform image2D spvRWriteImages2D_heapResourceHeap[];
+layout(set = 1, binding = 2, r32f) readonly uniform image3D spvRWriteImages3D_heapResourceHeap[];
+layout(set = 1, binding = 2) uniform image2D spv_164ResourceHeap[];
 layout(set = 1, binding = 2, std140) uniform UBO140
 {
     float data[2];
-} spvUBO140_189ResourceHeap[];
+} spvUBO140_186ResourceHeap[];
 
 layout(set = 1, binding = 2, std140) uniform UBO430
 {
     vec3 data[2];
-} spvUBO430_203ResourceHeap[];
+} spvUBO430_201ResourceHeap[];
 
 layout(set = 1, binding = 2, scalar) uniform UBOScalar
 {
     vec3 data[2];
-} spvUBOScalar_217ResourceHeap[];
+} spvUBOScalar_215ResourceHeap[];
 
-layout(set = 1, binding = 2, std430) readonly buffer SSBOReadOnly
+layout(set = 1, binding = 2, std430) readonly buffer SSBOReadOnlyNoWrite
 {
     vec4 data;
-} spvSSBOReadOnly_232ResourceHeap[];
+} spvSSBOReadOnlyNoWrite_230ResourceHeap[];
 
-layout(set = 1, binding = 2, std430) coherent writeonly buffer SSBOWriteOnly
+layout(set = 1, binding = 2, std430) coherent writeonly buffer SSBOWriteOnlyNoRead
 {
     vec4 data;
-} spvSSBOWriteOnly_246ResourceHeap[];
+} spvSSBOWriteOnlyNoRead_245ResourceHeap[];
 
 layout(set = 1, binding = 2, std430) buffer SSBO
 {
     vec4 data;
-} spvSSBO_255ResourceHeap[];
+} spvSSBO_254ResourceHeap[];
 
-layout(set = 1, binding = 2, r32ui) uniform uimage2D spv_263ResourceHeap[];
-layout(set = 1, binding = 2) uniform accelerationStructureEXT spv_278ResourceHeap[];
+layout(set = 1, binding = 2, r32ui) uniform uimage2D spv_259ResourceHeap[];
+layout(set = 1, binding = 2) uniform accelerationStructureEXT spv_274ResourceHeap[];
 
 void main()
 {
     FragColor = vec4(0.0);
     int desc_index = int(gl_FragCoord.x);
-    FragColor += texelFetch(spv_28ResourceHeap[nonuniformEXT(desc_index + 0)], 0, 0);
-    FragColor += texelFetch(spv_40ResourceHeap[nonuniformEXT(desc_index + 1)], ivec2(0), 0);
-    FragColor += texelFetch(spv_53ResourceHeap[nonuniformEXT(desc_index + 2)], ivec3(0), 0);
-    FragColor += texture(nonuniformEXT(sampler2D(spv_40ResourceHeap[int(gl_FragCoord.x)], spv_75SamplerHeap[int(gl_FragCoord.y)])), vec2(0.0), 0.0);
-    FragColor += vec4(texture(nonuniformEXT(sampler2DShadow(spv_40ResourceHeap[int(gl_FragCoord.x)], spv_75SamplerHeap[int(gl_FragCoord.y)])), vec3(vec3(0.0).xy, 0.0), 0.0));
-    imageStore(spv_114ResourceHeap[nonuniformEXT(desc_index + 3)], 0, FragColor);
-    imageStore(spv_123ResourceHeap[nonuniformEXT(desc_index + 4)], ivec2(0), FragColor);
-    imageStore(spv_132ResourceHeap[nonuniformEXT(desc_index + 5)], ivec3(0), FragColor);
-    FragColor += imageLoad(spv_114ResourceHeap[nonuniformEXT(desc_index + 6)], 0);
-    FragColor += imageLoad(spv_123ResourceHeap[nonuniformEXT(desc_index + 7)], ivec2(0));
-    FragColor += imageLoad(spv_132ResourceHeap[nonuniformEXT(desc_index + 8)], ivec3(0));
-    FragColor += imageLoad(spv_171ResourceHeap[nonuniformEXT(desc_index + 9)], ivec2(0));
-    int _181 = desc_index + 10;
-    FragColor += vec4(spvUBO140_189ResourceHeap[nonuniformEXT(_181)].data[1]);
-    int _197 = desc_index + 11;
-    FragColor += vec4(spvUBO430_203ResourceHeap[nonuniformEXT(_197)].data[1].x);
-    int _211 = desc_index + 12;
-    FragColor += vec4(spvUBOScalar_217ResourceHeap[nonuniformEXT(_211)].data[1].x);
-    int _225 = desc_index + 13;
-    FragColor += spvSSBOReadOnly_232ResourceHeap[nonuniformEXT(_225)].data;
+    FragColor += texelFetch(spv_25ResourceHeap[nonuniformEXT(desc_index + 0)], 0, 0);
+    FragColor += texelFetch(spv_36ResourceHeap[nonuniformEXT(desc_index + 1)], ivec2(0), 0);
+    FragColor += texelFetch(spv_49ResourceHeap[nonuniformEXT(desc_index + 2)], ivec3(0), 0);
+    FragColor += texture(nonuniformEXT(sampler2D(spv_36ResourceHeap[int(gl_FragCoord.x)], spv_68SamplerHeap[int(gl_FragCoord.y)])), vec2(0.0), 0.0);
+    FragColor += vec4(texture(nonuniformEXT(sampler2DShadow(spv_36ResourceHeap[int(gl_FragCoord.x)], spv_68SamplerHeap[int(gl_FragCoord.y)])), vec3(vec3(0.0).xy, 0.0), 0.0));
+    imageStore(spvWriteImages1D_heapResourceHeap[nonuniformEXT(desc_index + 3)], 0, FragColor);
+    imageStore(spvWriteImages2D_heapResourceHeap[nonuniformEXT(desc_index + 4)], ivec2(0), FragColor);
+    imageStore(spvWriteImages3D_heapResourceHeap[nonuniformEXT(desc_index + 5)], ivec3(0), FragColor);
+    FragColor += imageLoad(spvRWriteImages1D_heapResourceHeap[nonuniformEXT(desc_index + 6)], 0);
+    FragColor += imageLoad(spvRWriteImages2D_heapResourceHeap[nonuniformEXT(desc_index + 7)], ivec2(0));
+    FragColor += imageLoad(spvRWriteImages3D_heapResourceHeap[nonuniformEXT(desc_index + 8)], ivec3(0));
+    FragColor += imageLoad(spv_164ResourceHeap[nonuniformEXT(desc_index + 9)], ivec2(0));
+    int _180 = desc_index + 10;
+    FragColor += vec4(spvUBO140_186ResourceHeap[nonuniformEXT(_180)].data[1]);
+    int _196 = desc_index + 11;
+    FragColor += vec4(spvUBO430_201ResourceHeap[nonuniformEXT(_196)].data[1].x);
+    int _210 = desc_index + 12;
+    FragColor += vec4(spvUBOScalar_215ResourceHeap[nonuniformEXT(_210)].data[1].x);
+    int _226 = desc_index + 13;
+    FragColor += spvSSBOReadOnlyNoWrite_230ResourceHeap[nonuniformEXT(_226)].data;
     int _239 = desc_index + 14;
-    spvSSBOWriteOnly_246ResourceHeap[nonuniformEXT(_239)].data = vec4(20.0);
+    spvSSBOWriteOnlyNoRead_245ResourceHeap[nonuniformEXT(_239)].data = vec4(20.0);
     int _250 = desc_index + 15;
-    FragColor += spvSSBO_255ResourceHeap[nonuniformEXT(_250)].data;
-    uint _271 = imageAtomicAdd(spv_263ResourceHeap[nonuniformEXT(desc_index + 16)], ivec2(0), 50u);
-    rayQueryInitializeEXT(rq, spv_278ResourceHeap[nonuniformEXT(desc_index + 17)], 0u, 0u, vec3(0.0), 0.0, vec3(1.0, 0.0, 0.0), 1.0);
+    FragColor += spvSSBO_254ResourceHeap[nonuniformEXT(_250)].data;
+    uint _270 = imageAtomicAdd(spv_259ResourceHeap[nonuniformEXT(desc_index + 16)], ivec2(0), 50u);
+    rayQueryInitializeEXT(rq, spv_274ResourceHeap[nonuniformEXT(desc_index + 17)], 0u, 0u, vec3(0.0), 0.0, vec3(1.0, 0.0, 0.0), 1.0);
 }
 
